vuu-ui/packages/vuu-ui-controls/src/list/useList.ts

```diff
@@ -84,17 +84,12 @@ export const useList = <Item, S extends SelectionStrategy>({
     [dataHook, onSelect]
   );
 
-  const scrollContainer = useMemo<RefObject<HTMLElement>>(() => {
-    if (scrollContainerRef) {
-      return scrollContainerRef;
-    } else {
-      return {
-        current:
-          containerRef.current?.querySelector(".vuuList-scrollContainer") ??
-          null,
-      };
+  const setContainerRef = useCallback<RefCallback<HTMLDivElement>>((el) => {
```

Comment:
This is the crux of the List navigation fix. The problem was that some of the navigation functions rely on the containerRef being set. Because of the way list renders, a child of MeasuredItem, the containerRef , defined as a regular RefObject, was only set on second render. If navigation happens immediately, the ref was not yet set.

By using a RefCallback, we ensure that the refs are set immediately the List is rendered.